core/chains/evm/gas/models.go

```diff
@@ -380,11 +360,44 @@ func bumpDynamicFee(cfg Config, lggr logger.SugaredLogger, currentTipCap, curren
 	return DynamicFee{FeeCap: bumpedFeeCap, TipCap: bumpedTipCap}, nil
 }
 
-func getMaxGasPrice(userSpecifiedMax *assets.Wei, config Config) *assets.Wei {
-	return assets.WeiMin(config.EvmMaxGasPriceWei(), userSpecifiedMax)
+func bumpFeePrice(originalFeePrice *assets.Wei, feeBumpPercent uint16, feeBumpUnits *assets.Wei) *assets.Wei {
+	bumpedFeePrice := assets.MaxWei(
+		originalFeePrice.AddPercentage(feeBumpPercent),
+		originalFeePrice.Add(feeBumpUnits),
+	)
+	return bumpedFeePrice
 }
 
-func capGasPrice(calculatedGasPrice, userSpecifiedMax *assets.Wei, config Config) *assets.Wei {
-	maxGasPrice := getMaxGasPrice(userSpecifiedMax, config)
+func updateCurrentFeePrice(lggr logger.SugaredLogger, currentFeePrice *assets.Wei, bumpedFeePrice, maxGasPrice *assets.Wei) *assets.Wei {
+	if currentFeePrice != nil {
+		if currentFeePrice.Cmp(maxGasPrice) > 0 {
+			// Shouldn't happen because the estimator should not be allowed to
+			// estimate a higher gas than the maximum allowed
+			lggr.AssumptionViolationf("Ignoring current gas price of %s that would exceed max gas price of %s", currentFeePrice.String(), maxGasPrice.String())
+		} else if bumpedFeePrice.Cmp(currentFeePrice) < 0 {
+			// If the current gas price is higher than the old price bumped, use that instead
+			bumpedFeePrice = currentFeePrice
+		}
+	}
+	return bumpedFeePrice
+}
+
+func getMaxGasPrice(userSpecifiedMax *assets.Wei, maxGasPriceWei *assets.Wei) *assets.Wei {
+	return assets.WeiMin(userSpecifiedMax, maxGasPriceWei)
+}
+
+func capGasPrice(calculatedGasPrice, userSpecifiedMax *assets.Wei, maxGasPriceWei *assets.Wei) *assets.Wei {
+	maxGasPrice := getMaxGasPrice(userSpecifiedMax, maxGasPriceWei)
 	return assets.WeiMin(calculatedGasPrice, maxGasPrice)
 }
```

Comment:
these comparisons could probably be made chain-agnostic with generic comparisons?

Reply:
good idea! Changing them to `txmgrtypes.Fee` instead.